design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,116 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low
+
+## Proposed metrics for Rook to collect
+We propose naming metrics to indicate a hierarchy that can be parsed to add it to the Ceph telemetry
+collection in a more ordered fashion.
+
+For example `rook/kubernetes/version` would should be put into a JSON structure like shown:
+```json
+"rook": {
+  "kubernetes": {
+    "version"
+  }
+}
+```
+
+### Overall metrics
+- `rook-version` - Rook version.
+  - **DISCUSS:** should we add `rook/version` or rename this to maintain a consistent hierarchy?
+- `rook/kubernetes/version` - Kubernetes version.
+  - Ceph already collects os/kernel versions.
+- `rook/kubernetes/node-count` - Number of Kubernetes nodes.
+- `rook/csi/version` - Ceph CSI version.
+- Number of CSI plugin pods running - This gives idea of how many k8s nodes are in the k8s cluster that
+  are allowed to use Ceph storage.
+  - It is easiest count plugin pods separately since each DaemonSet is a different label. Plus, it
+    can show if some CSI drivers are ever disabled.
+  - `rook/csi/rbd-plugin-pod-count`
+  - `rook/csi/cephfs-plugin-pod-count`
+  - `rook/csi/nfs-plugin-pod-count`
+
+### CephCluster metrics
+- `rook/cluster/nodes-with-ceph-pods-count` - Number of nodes on which the cluster has running pods.
```

Comment:
Another metric to consider:
- `rook/cluster/mon/max-id` - The highest mon id, which increases as mons failover